### PR TITLE
Add --user flag to "secrets" and "envs"

### LINKED ISSUE #76: Create foundation for performing bulk actions
Allow admins to give a `--user` flag that specifies which users's resources should be the target of crud operations. Example:
```
$ coder secrets --user charlie@coder.com create mysql-password --from-literal 123password
```

Related: https://github.com/cdr/enterprise/pull/4497

### PR DESCRIPTION
**Pull request by cmoog**
## Samples
Setting a secret value for all users. Of course, were the admin to want a different value for each user, they would need to have some additional way of correlating the email to a secret prior to creating the Coder secret.
```bash
#!/bin/bash

# get all user emails
emails=$(coder users ls --output json | jq -c -r '.[] | .email')

for user in $emails
do
	# create a secret for that user
	coder secrets --user $user create mysql-password --from-literal password123 
done
```

Closes #76

### CODE REVIEW

File: cmd/coder/auth.go [hunk 1]
```diff
@@ -11,15 +11,15 @@ import (
 )
 
 // requireAuth exits the process with a nonzero exit code if the user is not authenticated to make requests
-func requireAuth() *entclient.Client {
-	client, err := newClient()
+func requireAuth(user string) *entclient.Client {
```
Comment (deansheather) [inline cmd/coder/auth.go:14]: How come you added the user to create the secret for to the struct? Wouldn't the API be better if it was just a param to the create secret function? Adding the user to the struct implies it's a sort of sudo thing like gitlab.

Comment (cmoog) [reply cmd/coder/auth.go:14]: My understanding is that we'll be expanding this concept to nearly all resources in the platform such that an admin could create/start/stop and environment for a specific user. So my thinking is that we'd want to parameterize the target user at the API client abstraction level.

Comment (deansheather) [reply cmd/coder/auth.go:14]: My issue with this however is that users of the API will find it tedious to have to recreate a new entclient every time if they wanted to do a Go-based solution for creating a secret for every user. You'd have to make an admin client for getting the user list, then a new client for every user.

How do other APIs handle sudo? I would've thought most APIs would do it like:
```go
type RequestOptions struct {
  Sudo string
}

var DefaultRequestOptions = &RequestOptions{
  Sudo: "me",
}

func (c *Client) GetSomething(ctx context.Context, opts *RequestOptions) {
  if (opts == nil) {
    opts = DefaultRequestOptions
  }

  ...
}
```

Comment (cmoog) [reply cmd/coder/auth.go:14]: Ah, this is a good point. Changing now.

**Pull request merged**